lua/gitblame/init.lua

```diff
@@ -210,6 +206,52 @@ local function get_blame_info(filepath, linenumber)
     return info
 end
 
+---@param filepath string
+---@param line1 number
+---@param line2 number
+---@return table<number,BlameInfo>
+local function get_range_blame_info(filepath, line1, line2)
+    ---@type table<number,BlameInfo>
+    local range_info = {}
+
+    for _, blame in ipairs(files_data[filepath].blames) do
+        local blame_is_out_of_range =
+            (blame.startline < line1 and blame.endline < line1) or
+            (blame.startline > line2 and blame.endline > line2)
+
+        if not blame_is_out_of_range then
+            range_info[#range_info + 1] = blame
+        end
+    end
```

Comment:
Instead of looping over the range of lines and looping over the range of blames for each line, it's more performant (`O(n)`) to loop once over blames and extract the blames that overlap the range of lines